src/Compilers/Server/VBCSCompiler/NamedPipeClientConnection.cs

```diff
@@ -229,6 +202,44 @@ internal NamedPipeClientConnection(ICompilerServerHost compilerServerHost, strin
             }
         }
 
+        protected override void ValidateBuildRequest(BuildRequest request)
```

Comment:
I'm not sure I can test this given our current Jenkins setup.  In order to fail this test I need the client and server pipe to have different identities.  Essentially I need to have two users accounts available.  

Let me chat with Andy when I get back in town.  I'll see if we can devise a way to get some coverage here.  Even if it's only in specific runs where multiple accounts are made available to the xunit process.